src/services/cohortCreation/index.ts

```diff
@@ -64,49 +84,76 @@ const createCohortGroup = async (jsonQuery: string): Promise<IGroup> => {
         return `/Patient?${patientFilter}&${criteria.filterSolr}`
       case 'Condition':
         return `/Patient?${patientFilter}&_has:Condition:patient:${criteria.filterSolr}`
+      case 'Composition':
+        return `/DocumentReference/$regex?${criteria.filterSolr?.replace(/^_text/, 'regex')}`
       default:
         return ''
     }
   }
 
-  // We record every requests needed to fetch the patients that meet the cohort criteria
-  // and then we fetch the patients once
-  const queries = criteriaGroup.reduce<string[]>((acc, group) => {
-    return [...acc, ...group.criteria.reduce<string[]>((acc, criteria) => [...acc, queryByResourceType(criteria)], [])]
-  }, [])
-  await Promise.all(queries.map((query) => getPatients(query)))
-
-  const patientIds = await criteriaGroup
-    .sort((a, b) => Number(b.isInclusive) - Number(a.isInclusive))
-    .reduce(async (patientIdsAcc, group) => {
-      const groupPatientIds = await group.criteria
+  const criteriaGroup = query.request[0].criteria
+
+  // We record every requests needed to fetch the patients/documents that meet the cohort criteria
+  // and then we fetch the patients/documents once
+  const { patientQueries, documentQueries } = criteriaGroup.reduce<{
+    patientQueries: string[]
+    documentQueries: string[]
+  }>(
+    (acc, group) => {
+      return group.criteria.reduce<{
```

Comment:
nit: don't need to use return here

Reply:
my ide complain for some reason if I don't

Reply:
ok